app/src/views/private/components/comments-sidebar-detail/comment-input.vue

```diff
@@ -70,237 +79,268 @@ import { User } from '@directus/shared/types';
-				countAfter++;
-			}
+		userPreviews.value = newUsers;
 
-			const before = text.substring(0, countBefore + (text.charAt(countBefore) === '\n' ? 1 : 0));
-			const after = text.substring(countAfter);
+		searchResult.value = result.data.data;
+	} catch (e) {
+		return e;
+	}
+}, 200);
+
+function cancel() {
+	if (props.existingComment) {
+		emit('cancel');
+	} else {
+		newCommentContent.value = '';
+		focused.value = false;
+	}
+}
 
-			newCommentContent.value = before + ' @' + user.id + after;
-		}
+// Why are selections so weird?
+function saveCursorPosition() {
+	if (document.getSelection) {
+		const selection = document.getSelection();
+		if (selection) {
+			lastCaretOffset = selection.anchorOffset;
 
-		function triggerSearch({ searchQuery, caretPosition }: { searchQuery: string; caretPosition: number }) {
-			triggerCaretPosition = caretPosition;
+			const range = selection?.getRangeAt(0);
+			range?.setStart(commentElement.value?.$el, 0);
+			lastCaretPosition = range?.cloneContents().textContent?.length ?? 0;
 
-			showMentionDropDown.value = true;
-			loadUsers(searchQuery);
-			selectedKeyboardIndex.value = 0;
+			selection.removeAllRanges();
 		}
+	}
+}
 
-		function avatarSource(url: string) {
-			if (url === null) return '';
-			return addTokenToURL(getRootPath() + `assets/${url}?key=system-small-cover`);
-		}
+function insertAt() {
+	saveCursorPosition();
+	document.getSelection()?.removeAllRanges();
+	insertText(' @');
+}
 
-		async function postComment() {
-			if (newCommentContent.value === null || newCommentContent.value.length === 0) return;
-			saving.value = true;
-
-			try {
-				if (props.existingComment) {
-					await api.patch(`/activity/comment/${props.existingComment.id}`, {
-						comment: newCommentContent.value,
-					});
-				} else {
-					await api.post(`/activity/comment`, {
-						collection: props.collection,
-						item: props.primaryKey,
-						comment: newCommentContent.value,
-					});
-				}
-
-				props.refresh();
-
-				newCommentContent.value = '';
-
-				notify({
-					title: t('post_comment_success'),
-				});
-			} catch (err: any) {
-				unexpectedError(err);
-			} finally {
-				saving.value = false;
-			}
-		}
+function insertText(text: string) {
+	if (newCommentContent.value === null) {
+		lastCaretPosition = 0;
+		newCommentContent.value = '';
+	}
 
-		function pressedUp() {
-			if (selectedKeyboardIndex.value > 0) {
-				selectedKeyboardIndex.value--;
-			}
-		}
+	newCommentContent.value = [
+		newCommentContent.value.slice(0, lastCaretPosition),
+		text,
+		newCommentContent.value.slice(lastCaretPosition),
+	].join('');
+
+	setTimeout(() => {
+		commentElement.value?.$el.focus();
+		document.getSelection()?.setPosition(document.getSelection()?.anchorNode ?? null, lastCaretOffset + text.length);
+
+		const inputEvent = new Event('input', { bubbles: true });
+		commentElement.value?.$el.dispatchEvent(inputEvent);
+	}, 10);
```

Comment:
Is this needed? Having to rely on a 10ms timeout for something feels like a hack

Reply:
What does this section do?

Reply:
This sections resets the focus to the right position in the comment. We have to wait 10ms because smaller values won't focus the comment area after inserting an emoji.
I would love to give you an explanation to why this is like that but I don't even know why we have to use a setTimeout here. 😄

Reply:
My best guess is that we have to wait until vue updates the dom with the inserted value.